Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -298,15 +300,18 @@ private extension UInt8 {
         case .keepAlive:
             return true
         case .unknown:
-            guard let connection = self["connection"].first?.lowercased() else {
-                // HTTP 1.1 use keep-alive by default if not otherwise told.
-                return version.major == 1 && version.minor == 1
-            }
-            
-            if connection == "close" {
-                return false
+            if !self.headers.isEmpty {
```

Comment:
this seems superfluous as you iterate `self.headers` just after that. If this is empty, the loop won't do anything either.

Reply:
ok :)